src/pQueue.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+export default class pQueue {
+  constructor() {
+    this.q = []
+  }
+
+  enqueue(element, priority){
+    // push an object to the end of the arrat
+    // sort the array according to priority
+    this.q[this.q.length] = {element, priority}
+
+    var sorted = this.q.sort(function(a,b){
+      return parseFloat(b.priority) - parseFloat(a.priority)
```

Comment:
why not `parseFloat` the priority when you add it to the array so you dont have to do it a ton of tines during the sort?